File: Resources/TOI-Zero_68/A2/005/005.cpp
#include<bits/stdc++.h>
using namespace std;

#define ll long long

int main(){
	ios::sync_with_stdio(0);
	cin.tie(0);
	int w,h;
	cin >> w >> h;
	int m,n;
	cin >> m >> n;
	vector<int> x(m+2),y(n+2);
	x[0]=0,x[m+1]=w;
	y[0]=0,y[n+1]=h;
	for(int i=1;i<=m;i++){
		cin >> x[i];
	}
	for(int i=1;i<=n;i++){
		cin >> y[i];
	}
	vector<int> widths,heights;
	for(int i=0;i<m+1;i++){
		widths.push_back(x[i+1]-x[i]);
	}
	for(int i=0;i<n+1;i++){
		heights.push_back(y[i+1]-y[i]);
	}
	vector<int> areas;
	for(int width : widths){
		for(int height : heights){
			areas.push_back(width*height);
		}
	}
	sort(areas.rbegin(),areas.rend());
	cout << areas[0] << " " << (areas.size()>1 ? areas[1]:areas[0]);
}

File: Resources/Projects/Archive/2023/Competition/Programming/Contest/TOI-Zero_68/A2/019/TOI-Zero_68/A2/005/005.py
w, h = map(int, input().split())
m, n = map(int, input().split())

x = [0] * (m + 2)
y = [0] * (n + 2)
x[0] = 0
x[m + 1] = w
y[0] = 0
y[n + 1] = h

for i in range(1, m + 1):
    x[i] = int(input())
for i in range(1, n + 1):
    y[i] = int(input())

widths = []
heights = []

for i in range(m + 1):
    widths.append(x[i + 1] - x[i])
for i in range(n + 1):
    heights.append(y[i + 1] - y[i])

areas = []
for width in widths:
    for height in heights:
        areas.append(width * height)

areas.sort(reverse=True)
print(areas[0], end=" ")
if len(areas) > 1:
    print(areas[1])
else:
    print(areas[0]) 